Tests/ML/histopathology/models/test_deepmil.py

```diff
@@ -157,6 +160,106 @@ def test_lightningmodule_mean_pooling(
+            TilesDataset.TILE_ID_COLUMN: [f"{slide_idx}-{tile_idx}"
+                                          for tile_idx in range(bag_size)],
+            TilesDataset.IMAGE_COLUMN: rand(bag_size, *input_dim),
+            TilesDataset.LABEL_COLUMN: bag_label.expand(bag_size),
+        }
+        sample[TilesDataset.PATH_COLUMN] = [tile_id + '.png'
+                                            for tile_id in sample[TilesDataset.TILE_ID_COLUMN]]
+        bags.append(sample)
+    batch = default_collate(bags)
+
+    # ================
+    # Test that the module metrics match manually computed metrics with the correct inputs
+    module_metrics_dict = module.test_metrics
+    independent_metrics_dict = module.get_metrics()
+
+    # Patch the metrics to check that the inputs are valid. In particular, test that the scores
+    # do not have integral values, which would suggest that hard labels were passed instead.
+    for metric_obj in module_metrics_dict.values():
+        metric_obj.update = add_callback(metric_obj.update, validate_metric_inputs)
+
+    results = module.test_step(batch, 0)
+    predicted_probs = results[ResultsKey.PROB]
+    true_labels = results[ResultsKey.TRUE_LABEL]
+
+    for key, metric_obj in module_metrics_dict.items():
+        value = metric_obj.compute()
+        expected_value = independent_metrics_dict[key](predicted_probs, true_labels)
+        assert torch.allclose(value, expected_value), f"Discrepancy in '{key}' metric"
+
+    # ================
+    # Test that thresholded metrics (e.g. accuracy, precision, etc.) change as the threshold is varied.
+    # If they don't, it suggests the inputs are hard labels instead of continuous scores.
+    thresholded_metrics_keys = [key for key, metric in module_metrics_dict.items()
+                                if hasattr(metric, 'threshold')]
+
+    def set_metrics_threshold(metrics_dict: Any, threshold: float) -> None:
+        for key in thresholded_metrics_keys:
+            metrics_dict[key].threshold = threshold
+
+    def reset_metrics(metrics_dict: Any) -> None:
+        for metric_obj in metrics_dict.values():
+            metric_obj.reset()
+
+    low_threshold, high_threshold = torch.quantile(predicted_probs, torch.tensor([0.1, 0.9]))
+
+    reset_metrics(module_metrics_dict)
+    set_metrics_threshold(module_metrics_dict, threshold=low_threshold)
+    _ = module.test_step(batch, 0)
+    results_low_threshold = {key: module_metrics_dict[key].compute()
+                             for key in thresholded_metrics_keys}
+
+    reset_metrics(module_metrics_dict)
+    set_metrics_threshold(module_metrics_dict, threshold=high_threshold)
+    _ = module.test_step(batch, 0)
+    results_high_threshold = {key: module_metrics_dict[key].compute()
+                              for key in thresholded_metrics_keys}
+
+    for key in thresholded_metrics_keys:
+        assert not torch.allclose(results_low_threshold[key], results_high_threshold[key]), \
```

Comment:
the default tolerance is quite low, can make sense to pass it a larger number given here we're interested in checking these results ARE different?